src/PyNvCodec/src/PyNvDecoder.cpp

```diff
@@ -820,19 +823,126 @@ void Init_PyNvDecoder(py::module& m)
            R"pbdoc(
         Return dictionary with Nvdec capabilities.
     )pbdoc")
-      .def(
-          "DecodeSingleSurface",
-          [](shared_ptr<PyNvDecoder> self, PacketData& out_pkt_data) {
-            DecodeContext ctx(nullptr, nullptr, nullptr, &out_pkt_data, nullptr,
-                              false);
-            if (self->DecodeSurface(ctx))
-              return ctx.GetSurfaceMutable();
-            else
-              return make_empty_surface(self->GetPixelFormat());
-          },
-          py::arg("pkt_data"), py::return_value_policy::take_ownership,
-          py::call_guard<py::gil_scoped_release>(),
-          R"pbdoc(
+    .def(
+      "DecodeSurfaceFromPacket",
+      [](shared_ptr<PyNvDecoder> self, PacketData& in_pkt_data,
+             py::array_t<uint8_t>& packet, PacketData& out_pkt_data,
+             bool bOutputNVCVImage) -> py::object {
```

Comment:
`bOutputNVCVImage` shouldn't be necessary. If you want the users to decide on the output then a different function name should be used. For instance, the existing API already has
```
DecodeFrameFromPacket
DecodeSurfaceFromPacket
```
so
```
DecodeNcvImageFromPacket
```
would make sense to me, because the returned value is not a `Surface`